packages/button/docs/Button.stories.tsx

```diff
@@ -110,6 +111,19 @@ const StickerSheetTemplate: Story<{ isReversed: boolean }> = ({
     iconPosition: "end",
   }
 
+  const BADGE_LEFT_PROPS: ButtonProps = {
+    label: "Label",
+    icon: filterIcon,
+    badge: { text: "4" },
+  }
+
+  const BADGE_RIGHT_PROPS: ButtonProps = {
```

Comment:
Looking at Figma, there's probably no use case for a badge with an icon on the right. I'd err away from an example of something people shouldn't be doing (especially since the gap looks wrong).

I do wonder if there's a case when we'd want a badge and no icon though (a design question).

Reply:
There is a use case for this! It was actually what sparked this PR. Elliot brought it to mine attention. It used in survey results

Reply:
![image](https://user-images.githubusercontent.com/48232362/209017338-e55556d4-8811-4247-a50c-2831eec06cd9.png)

Reply:
Hmm interesting... Do you know if there's any case where we would have a badge but not an icon? Would we need an example for that too?

But fair enough to show this case since there's a real scenario for it. The spacing looks off to me though, but happy for that to be fixed in a separate PR :)